flex-squid/src/main/java/org/sonar/flex/metrics/FileMetrics.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * SonarQube Flex Plugin
+ * Copyright (C) 2010-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.flex.metrics;
+
+import com.sonar.sslr.api.AstNode;
+import java.util.Set;
+import org.sonar.flex.FlexGrammar;
+import org.sonar.flex.FlexVisitorContext;
+
+public class FileMetrics {
+
+  private final int numberOfStatements;
+  private final int numberOfClasses;
+  private final int numberOfFunctions;
+  private final FileLinesVisitor fileLinesVisitor = new FileLinesVisitor();
+
+  public FileMetrics(FlexVisitorContext context) {
+    AstNode rootTree = context.rootTree();
+    numberOfStatements = rootTree.getDescendants(
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "rootTree" is nullable here. ([squid:S2259](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.flex%3Aflex&branch=SONARFLEX-105&issues=AWGd6bBDR7xclXvFe_Id&open=AWGd6bBDR7xclXvFe_Id)

[ISSUE_KEY]: AWGd6bBDR7xclXvFe_Id